src/web/mjs/connectors/ZeroScans.mjs

```diff
@@ -1,15 +1,80 @@
-import Genkan from './templates/Genkan.mjs';
-
-export default class ZeroScans extends Genkan {
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+export default class ZeroScans extends Connector {
 
     constructor() {
         super();
         super.id = 'zeroscans';
         super.label = 'ZeroScans';
         this.tags = [ 'manga', 'high-quality', 'english', 'scanlation' ];
         this.url = 'https://zeroscans.com';
-        this.links = {
-            login: 'https://zeroscans.com/login'
+
+        this.config = {
+            quality: {
+                label: 'Preferred quality',
+                description: 'The preferred quality of the chapters',
+                input: 'select',
+                options: [
+                    { value: 'good_quality', name: 'Good Quality' },
+                    { value: 'high_quality', name: 'High Quality' }
+                ],
+                value: 'high_quality'
+            }
         };
     }
+
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri, this.requestOptions);
+        const script = `
+            new Promise(resolve => {
+                resolve(JSON.stringify(window.__ZEROSCANS__));
+            });
+        `;
+        const { data } = await Engine.Request.fetchUI(request, script);
+        const details = data[0].details;
+        return new Manga(this, `${details.id}_${details.slug}`, details.name.trim());
```

Comment:
Is it safe to assume that the sperator `_` will not appear in `details.id` or `details.slug`?

Reply:
details.id is a number, while details.slug is a string with hyphens instead of spaces.

I put both of them into the manga ID because the API request for chapters needs details.id, while the API request for pages needs the slug.